src/main/java/bio/terra/cda/app/controller/QueryApiController.java

```diff
@@ -95,12 +96,36 @@ private String createNextUrl(String jobId, int offset, int limit) {
   }
 
   @Override
-  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body) {
-
+  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body, String system) {
     String table = applicationConfiguration.getBqTable() + "." + version;
     NestedColumn nt = NestedColumn.generate(body);
-    String querySql = "SELECT DISTINCT " + nt.getColumn() + " FROM " + table + nt.getUnnestClause();
-    logger.debug("uniqueValues: " + querySql);
+    Set<String> unnestClauses = nt.getUnnestClauses();
+    StringBuffer unnestConcat = new StringBuffer();
+    String whereClause = null;
+
+    if (system != null) {
+      NestedColumn whereColumns = NestedColumn.generate("ResearchSubject.identifier.system");
+      whereClause = " WHERE " + whereColumns.getColumn() + " = " + system;
+      // add any additional 'where' unnest partials that aren't already included in columns-unnest
+      // clauses
+      whereColumns.getUnnestClauses().stream().forEach((c) -> unnestClauses.add(c));
```

Comment:
I think this is the same as 
```suggestion
      unnestClauses.addAll(whereColumns.getUnnestClauses());
```

Reply:
Used suggestion.